src/Parbad.Gateway/PayIr/src/Internal/PayIrHelper.cs

```diff
@@ -91,7 +91,13 @@ public static PaymentVerifyResult CreateVerifyResult(string response, MessagesOp
                 return PaymentVerifyResult.Failed(message);
             }
 
-            return PaymentVerifyResult.Succeed(result.TransId, messagesOptions.PaymentSucceed);
+            var verifyResult = PaymentVerifyResult.Succeed(result.TransId, messagesOptions.PaymentSucceed);
```

Comment:
Create the PayIrVerifyAdditionalData class here and fill the properties and then add it inside the dictionary.
verifyResult.AdditionalData.Add(YourExtensionsClass.AdditionalDataKey] = class;

Reply:
AdditionalData is a <string,string> dictionary. Should I serialize the class?
What is the harm to add 4 items to the dictionary?

Reply:
That's right. The AdditionalData is still string,string. I didn't commit the new changes yet.
It's going to be a dictionary of <string, object> after I commit the new changes.
So for now let it be like this.